assets/src/edit-story/components/library/panes/media/mediaUtils.js

```diff
@@ -85,6 +85,140 @@ export const getResourceFromMediaPicker = (mediaPickerEl) => {
   };
 };
 
+/**
+ * Generates a virtual image to pick metadata resource
+ *
+ * @param {Object} image File object
+ * @return {Promise} Resource object
+ */
+
+const getImageMetadata = (image) => {
+  const reader = new window.FileReader();
+
+  return new Promise((resolve, reject) => {
+    reader.onload = function(e) {
+      const src = e.target.result;
+      const randomId = Math.floor(Date.now() / 1000);
+
+      const img = new window.Image();
+
+      img.onload = function() {
+        const width = img.width;
+        const height = img.height;
+
+        resolve({
+          type: 'image',
+          src,
+          width,
+          height,
+          mimeType: image.type,
+          posterId: randomId,
+          poster: '',
+          videoId: undefined,
+          local: true,
+          lengthFormatted: undefined,
+        });
+      };
+
+      img.src = src;
+    };
+
+    reader.onerror = reject;
+
+    reader.readAsDataURL(image);
+  });
+};
+
+/**
+ * Generates a virtual video to pick metadata resource
+ *
+ * @param {Object} video File object
+ * @return {Promise} Resource object
+ */
+
+const getVideoMetadata = (video) => {
+  const reader = new window.FileReader();
+
+  return new Promise((resolve, reject) => {
+    reader.onload = function() {
+      const blob = new window.Blob([reader.result], { type: video.type });
+      const src = URL.createObjectURL(blob);
+      const v = document.createElement('video');
+      const randomId = Math.floor(Date.now() / 1000);
+
+      const snapshot = () => {
+        const canvas = document.createElement('canvas');
+        canvas.width = v.videoWidth;
+        canvas.height = v.videoHeight;
+        canvas.getContext('2d').drawImage(v, 0, 0, canvas.width, canvas.height);
+        const poster = canvas.toDataURL();
+        const success = poster.length > 100000;
+
+        if (success) {
+          const img = new window.Image();
+
+          img.onload = function() {
+            const width = img.width;
+            const height = img.height;
+
+            resolve({
+              type: 'video',
+              src,
+              width,
+              height,
+              mimeType: video.type,
+              posterId: randomId,
+              poster,
+              videoId: randomId,
+              local: true,
+              lengthFormatted: undefined,
+            });
+          };
+
+          img.src = poster;
+        }
+        return success;
+      };
+      const timeupdate = () => {
+        if (snapshot()) {
+          v.removeEventListener('timeupdate', timeupdate);
+          v.pause();
+        }
+      };
+      v.addEventListener(
+        'loadeddata',
+        () => snapshot() && v.removeEventListener('timeupdate', timeupdate)
+      );
+      v.addEventListener('timeupdate', timeupdate);
+      v.preload = 'metadata';
+      v.src = src;
+      v.muted = true;
+      v.playsInline = true;
+      v.play();
```

Comment:
Do we need to play this? Or is it enough to wait for "can play"? Also, elsewhere we had to skip to 0.5 seconds or even more. A lot of "fade in" videos out there.